analysis/tests/lighttpd-minimal/src/main.rs

```diff
@@ -4,88 +4,1149 @@
+    fn http_response_write_header(r: *mut request_st);
+    fn http_response_handler(r: *mut request_st) -> handler_t;
+    fn plugins_call_handle_request_done(r: *mut request_st) -> handler_t;
+    fn plugins_call_handle_connection_accept(con: *mut connection) -> handler_t;
+    fn plugins_call_handle_connection_shut_wr(con: *mut connection) -> handler_t;
+    fn plugins_call_handle_connection_close(con: *mut connection) -> handler_t;
+    fn config_cond_cache_reset(r: *mut request_st);
+    fn sock_addr_cache_inet_ntop_copy_buffer(
+        b: *mut buffer,
+        saddr: *const sock_addr,
+    ) -> libc::c_int;
     fn calloc(_: libc::c_ulong, _: libc::c_ulong) -> *mut libc::c_void;
+    fn free(_: *mut libc::c_void);
+    fn close(__fd: libc::c_int) -> libc::c_int;
+    fn read(__fd: libc::c_int, __buf: *mut libc::c_void, __nbytes: size_t) -> ssize_t;
+    fn __errno_location() -> *mut libc::c_int;
 }
-
+pub type __int8_t = libc::c_schar;
+pub type __uint8_t = libc::c_uchar;
+pub type __uint16_t = libc::c_ushort;
+pub type __int32_t = libc::c_int;
+pub type __uint32_t = libc::c_uint;
+pub type __uint64_t = libc::c_ulong;
+pub type __dev_t = libc::c_ulong;
+pub type __uid_t = libc::c_uint;
+pub type __gid_t = libc::c_uint;
+pub type __ino_t = libc::c_ulong;
+pub type __mode_t = libc::c_uint;
+pub type __nlink_t = libc::c_ulong;
+pub type __off_t = libc::c_long;
+pub type __off64_t = libc::c_long;
+pub type __pid_t = libc::c_int;
+pub type __time_t = libc::c_long;
+pub type __clockid_t = libc::c_int;
+pub type __blksize_t = libc::c_long;
+pub type __blkcnt_t = libc::c_long;
+pub type __ssize_t = libc::c_long;
+pub type __syscall_slong_t = libc::c_long;
+pub type __socklen_t = libc::c_uint;
+pub type gid_t = __gid_t;
+pub type uid_t = __uid_t;
+pub type off_t = __off64_t;
+pub type pid_t = __pid_t;
+pub type ssize_t = __ssize_t;
+pub type clockid_t = __clockid_t;
+pub type time_t = __time_t;
+pub type size_t = libc::c_ulong;
+pub type int8_t = __int8_t;
+pub type int32_t = __int32_t;
 #[derive(Copy, Clone)]
 #[repr(C)]
-pub struct connection {
-    pub fd: libc::c_int,
-    pub fdn: *mut fdnode,
-    pub next: *mut connection,
-    pub prev: *mut connection,
+pub struct timespec {
+    pub tv_sec: __time_t,
+    pub tv_nsec: __syscall_slong_t,
 }
-
+pub type uint8_t = __uint8_t;
+pub type uint16_t = __uint16_t;
+pub type uint32_t = __uint32_t;
+pub type uint64_t = __uint64_t;
+pub type uintptr_t = libc::c_ulong;
+pub type unix_time64_t = time_t;
+pub type unix_timespec64_t = timespec;
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub struct server {
+    pub plugin_slots: *mut libc::c_void,
+    pub config_context: *mut array,
+    pub config_captures: libc::c_int,
     pub ev: *mut fdevents,
-    pub cur_fds: libc::c_int,
+    pub network_backend_write: Option<
+        unsafe extern "C" fn(libc::c_int, *mut chunkqueue, off_t, *mut log_error_st) -> libc::c_int,
+    >,
+    pub request_env: Option<unsafe extern "C" fn(*mut request_st) -> handler_t>,
```

Comment:
Is leaving these in types fine?

Reply:
yes, although I'm not sure what you're concerned about

Reply:
I just know we can't fully handle function pointers, so I wasn't sure if it's okay to have them in types and just not use them.

Reply:
that's correct, only a subset of types are iterated over (ADTs) and it's their use in MIR bodies that would pose an issue